src/SyncTasks.ts

```diff
@@ -266,30 +282,42 @@ module Internal {
             return this;
         }
 
-        resolve(obj?: T): Deferred<T> {
-            if (this._completedSuccess || this._completedFail) {
-                throw new Error('Already Completed');
-            }
-            this._completedSuccess = true;
-            this._storedResolution = obj;
+        resolve(obj?:  T): Deferred<T> {
+           this._checkState(true);
+           this._completedSuccess = true;
+           this._storedResolution = obj;
 
-            this._resolveSuccesses();
+           this._resolveSuccesses();
 
-            return this;
+           return this;
         }
 
         reject(obj?: any): Deferred<T> {
-            if (this._completedSuccess || this._completedFail) {
-                throw new Error('Already Completed');
-            }
-            this._completedFail = true;
-            this._storedErrResolution = obj;
+           this._checkState(false);
+           this._completedFail = true;
+           this._storedErrResolution = obj;
 
-            this._resolveFailures();
+           this._resolveFailures();
 
-            SyncTask._enforceErrorHandled(this);
+           SyncTask._enforceErrorHandled(this);
 
-            return this;
+           return this;
+        }
+
+        private _checkState(resolve: boolean) {
+            if (this._completedSuccess || this._completedFail) {
+                if (this._completeStack) {
+                    console.error(this._completeStack.message, this._completeStack.stack);
+                }
+
+                const message = 'Failed to ' + resolve ? 'resolve' : 'reject' +
+                 ' the task is already ' + this._completedSuccess ? 'resolved' : 'rejected';
```

Comment:
Suggestion: put a separator (e.g. ': the task...') in the message, so it reads 'Failed to resolve: the task is already resolved'.